File: Java/jakarta/collectionsync/src/main/java/com/ithit/webdav/samples/collectionsync/websocket/GetHttpSessionConfigurator.java
package com.ithit.webdav.samples.collectionsync.websocket;

import jakarta.websocket.HandshakeResponse;
import jakarta.websocket.server.HandshakeRequest;
import jakarta.websocket.server.ServerEndpointConfig;

import static com.ithit.webdav.samples.collectionsync.websocket.WebSocketServer.INSTANCE_HEADER_NAME;

public class GetHttpSessionConfigurator extends ServerEndpointConfig.Configurator {
    @Override
    public void modifyHandshake(ServerEndpointConfig config,
                                HandshakeRequest request,
                                HandshakeResponse response) {
        config.getUserProperties().put(INSTANCE_HEADER_NAME, request.getHeaders()
                .entrySet()
                .stream()
                .filter(x -> x.getKey().equalsIgnoreCase(INSTANCE_HEADER_NAME))
                .findFirst().map(x -> {
                    if (!x.getValue().isEmpty()) {
                        return x.getValue().get(0);
                    }
                    return "";
                })
                .orElse(""));
    }
}

File: Kotlin/filesystemstorage/src/main/kotlin/com/ithit/webdav/samples/fsstorageservlet/websocket/GetHttpSessionConfigurator.kt
package com.ithit.webdav.samples.fsstorageservlet.websocket

import javax.websocket.HandshakeResponse
import javax.websocket.server.HandshakeRequest
import javax.websocket.server.ServerEndpointConfig

class GetHttpSessionConfigurator : ServerEndpointConfig.Configurator() {
    override fun modifyHandshake(
        config: ServerEndpointConfig,
        request: HandshakeRequest,
        response: HandshakeResponse
    ) {
        config.userProperties[INSTANCE_HEADER_NAME] = request.headers
            .entries
            .stream()
            .filter { x -> x.key.equals(INSTANCE_HEADER_NAME, ignoreCase = true) }
            .findFirst().map { x ->
                if (x.value.isNotEmpty()) {
                    return@map x.value[0]
                }
                ""
            }
            .orElse("")
    }
}